src/directedGraph.js

```diff
@@ -0,0 +1,92 @@
+export default class DirectedGraph {
+  constructor(){
+    this.vertices = 0
+    this.edges = []
+    this.edgeCounter = 0
+  }
+  makeGraph(){
+    for( let i = 0; i < this.vertices; ++i ) {
+      this.edges[i] = []
+      this.edges[i].push(i)
+    }
+  }
+
+  addVertex(v) { ++this.vertices }
+  hasVertex(v) {
+    let k
+    for(let i = 0; i < this.edges.length; i++){
+      if(i === v){
+        return true
+      }
+    }
+    return false
+  }
+
+  addDirection(v1, v2, w) {
+    this.edges[v1].push({vertex1: v1, vertex2: v2, weight: w})
+    this.edgeCounter++
+  }
+  hasDirection(v1, v2) {
+    if(this.edges[v1] && this.edges[v2]){
+      for( let i = 0; i < this.edges[v1].length; i++) {
+        if(this.edges[v1][i].vertex === v2){
+          return true
+        }
+      }
+    }
+    return false
+  }
```

Comment:
It looks like your storing the entire vertex in the edge? One way I got past a lot of problems that arose from this, was instead, store a hash key to a given vertex, in the edges, and the use lookup functions / tables to get those vertices.